examples/wifi/requests_wifi_context_manager_basics.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-FileCopyrightText: 2024 DJDevon3
+# SPDX-License-Identifier: MIT
+# Updated for Circuit Python 9.0
+""" WiFi Context Manager Basics Example """
+
+import os
+
+import adafruit_connection_manager
+import wifi
+
+import adafruit_requests
+
+# Get WiFi details, ensure these are setup in settings.toml
+ssid = os.getenv("CIRCUITPY_WIFI_SSID")
+password = os.getenv("CIRCUITPY_WIFI_PASSWORD")
+
+# Initalize Wifi, Socket Pool, Request Session
+pool = adafruit_connection_manager.get_radio_socketpool(wifi.radio)
+ssl_context = adafruit_connection_manager.get_radio_ssl_context(wifi.radio)
+requests = adafruit_requests.Session(pool, ssl_context)
+rssi = wifi.radio.ap_info.rssi
+
+JSON_GET_URL = "https://httpbin.org/get"
+
+print(f"\nConnecting to {ssid}...")
+print(f"Signal Strength: {rssi}")
+try:
+    # Connect to the Wi-Fi network
+    wifi.radio.connect(ssid, password)
+except OSError as e:
+    print(f"❌ OSError: {e}")
+print("✅ Wifi!\n")
+
+print("-" * 40)
+
+# This method requires an explicit close
+print("Explicit Close() Example")
+response = requests.get(JSON_GET_URL)
+print(" | ✅ Connected to JSON")
+
+json_data = response.json()
```

Comment:
If I am understanding the conversation here correctly I think it makes the most sense to switch `reponse.json()` here to use `response.text()` instead. Then if you want you can use `json.loads` to parse it and get the data out.

That way the one that is illustrating the manual call to `close()` is serving it's intended purpose about illustrating the necessity to close afterward when the context processor isn't in use.

Reply:
It doesn't matter if I do `data = response.text` or `data = response.json` both are available after the explicit close because both are being buffered into the data variable.  Can you post an example of what you mean because I'm not understanding the difference in use.  Yes .text will print out the raw json data format and .json prints out a parsed version.  With regards to .close() I'm not seeing any difference in behavior because the data variable acts as a buffer.